ext/kernelparser/debian/debian.go

```diff
@@ -24,32 +24,32 @@ func init() {
 	kernelparser.RegisterParser("debian", parser)
 }
 
-func parser(_ database.Datastore, kernelVersion, osImage string) (*kernelparser.ParseMatch, bool, error) {
+func parser(_ database.Datastore, kernelVersion, osImage string) (*kernelparser.ParseMatch, error) {
 	if strings.Contains(osImage, "garden") {
 		return &kernelparser.ParseMatch{
 			Namespace:   gardenLinux,
 			Format:      format,
 			FeatureName: featureName,
 			Version:     kernelVersion,
-		}, true, nil
+		}, nil
 	}
 
 	if !strings.Contains(osImage, "debian") {
-		return nil, false, nil
+		return nil, kernelparser.ErrKernelUnrecognized
 	}
 
 	matches := regex.FindStringSubmatch(osImage)
 	if len(matches) == 0 {
-		return nil, false, fmt.Errorf("could not find Debian version in OS string: %q", osImage)
+		return nil, fmt.Errorf("could not find Debian version in OS string: %q", osImage)
```

Comment:
It means we will panic, right? Let's not panic when we cannot parse the OS, better handle it gracefully. Can we signal `ErrKernelUnrecognized`, for example? Or another status code?

Reply:
great catch i completely missed this!

Reply:
I removed the panic, and I added tests to the service to ensure we support these "non-standard" errors